crates/re_types_builder/src/codegen/cpp/mod.rs

```diff
@@ -696,24 +696,27 @@ impl QuotedObject {
             ));
         }
 
-        if are_types_disjoint(&obj.fields) {
-            // Implicit construct from the different variant types:
-            for obj_field in &obj.fields {
-                let snake_case_ident = format_ident!("{}", crate::to_snake_case(&obj_field.name));
-                let param_declaration =
-                    quote_variable(&mut hpp_includes, obj_field, &snake_case_ident);
+        if !obj.has_attr(ATTR_CPP_NO_FIELD_CTORS) {
+            if are_types_disjoint(&obj.fields) {
+                // Implicit construct from the different variant types:
+                for obj_field in &obj.fields {
+                    let snake_case_ident =
+                        format_ident!("{}", crate::to_snake_case(&obj_field.name));
+                    let param_declaration =
+                        quote_variable(&mut hpp_includes, obj_field, &snake_case_ident);
 
-                methods.push(Method {
+                    methods.push(Method {
                     docs: obj_field.docs.clone().into(),
                     declaration: MethodDeclaration::constructor(quote!(#pascal_case_ident(#param_declaration))),
                     definition_body: quote!(*this = #pascal_case_ident::#snake_case_ident(std::move(#snake_case_ident));),
                     inline: true,
                 });
+                }
```

Comment:
weird indentation

Reply:
maybe we need to simplify the surrounding code so that `cargo fmt` doesn't give up